guides/v2.1/test/integration/annotations/magentoDataFixture.md

```diff
@@ -0,0 +1,195 @@
+---
+group: integration-testing
+version: 2.1
+title: Data fixture annotation
+github_link: /test/integration/annotations/magentoDataFixture.md
+---
+
+Use data fixtures in integration tests to prepare the database for testing.
+The Integration Testing Framework reverts the database to its initial state automatically.
+A data fixture is a PHP script that sets data you want to reuse in your test.
+The script can be defined in a separate file or as a local method in the test case.
+To set up the date fixture, use the `@magentoDataFixture` annotation.
+
+## Format
+
+`@magentoDataFixture` takes an argument that points to the data fixture as a filename a local method.
+
+> Data fixture annotation
+
+```php?start_inline=1
+/**
+ * @magentoDataFixture <script_filename>|<method_name>
+ */
+ ```
+ 
+Here,
+ * `<script_filename>` is a filename of the PHP script
+ * `<method_name>` is a name of the method declared in the current class
+ 
+## Principles
+ 
+1. Do not use direct connection to a database in fixtures.
+It introduces a dependency on the database structure and potentially on the database vendor.
+
+2. Use the application (model) API to implement your data fixtures.
+ 
+## Usage
+ 
+Fixtures can be declared in two ways:
+
+- as a PHP script file that is used by other tests and test cases.
+- as a local method that is used by other tests in the test cases.
+ 
+### Fixture as a separate file
+ 
+Define the fixture in a separate file when you want to reuse it in different test cases.
+ 
+To declare the fixture, use the following convention:
+* a path that is relative to the `dev/tests/integration/<testsuite directory>`
+* containing only forward slashes `/` with no preceding slash. Example: `Magento/Core/Model/Layout/_files/layout_custom_handle.php`
+
+The ITF includes the declared PHP script to your test and executes it during test run.
+
+#### Example
+
+__Use case__: Verify the `getSortOrder()` method that gets the value of `sort_order`.
+
+Create a data fixture as a separate file called `layout_custom_handle.php` that sets `sort_order` to `456`.
+
+> The `layout_custom_handle.php` file content:
+
+```php?start_inline=1
+<?php
+$model = new Magento_Core_Model_Layout_Update();
```

Comment:
This is the class from Magento 1. Magento 2 `\Magento\Widget\Model\Layout\Update` instead

Reply:
changed to:
`$model = new \Magento\Widget\Model\Layout\Update();`